### PR TITLE
[CST] - Update Status Tab for the What We're Doing Section when feature flag cst_claim_phases is enabled 

### PR DESCRIPTION
**Pull request by pmclaren19**
## Summary

Updated `src/applications/claims-status/components/claim-status-tab/WhatWeAreDoing.jsx` so that when the feature flag `cst_claim_phases` is enabled the following occurs:
- The header text reads 'Step <CURRENT_STEP> of 8'  and the corresponding text from [this](https://docs.google.com/document/d/1WfWCfoQgKoMBg835VP7j_ptXmqb2o3RVme2u5dMOEr8/edit) doc 
- The description is the description listed for the given step in [this](https://docs.google.com/document/d/1WfWCfoQgKoMBg835VP7j_ptXmqb2o3RVme2u5dMOEr8/edit) doc 
- There is a new field that says 'Moved to this step on <PhaseChangeDate>
- The link text is 'Learn more about this step'

Added tests to `src/applications/claims-status/tests/components/claim-status-tab/WhatWeAreDoing.unit.spec.jsx`

Added new props for WhatWeAreDoing and updated `src/applications/claims-status/containers/ClaimStatusPage.jsx` with these new props

Added new helper functions to `helpers.js` to account for these new phases called `getClaimPhaseTypeHeaderText`, `getClaimPhaseTypeDescription`

Updated `src/applications/claims-status/tests/components/ClaimStatusPage.unit.spec.jsx` so that its referencing an actual phaseType

Added tests to `src/applications/claims-status/tests/utils/helpers.unit.spec.js`

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#82375

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

**Step 1**
Screenshot 2024-05-30 at 2 16 36 PM

**Step 2**
Screenshot 2024-05-30 at 1 29 26 PM

**Step 3**
Screenshot 2024-05-30 at 1 30 11 PM

**Step 4**
Screenshot 2024-05-30 at 1 30 39 PM

**Step 5** 
Screenshot 2024-05-30 at 1 31 18 PM

**Step 6**
Screenshot 2024-05-30 at 1 31 41 PM

**Step 7**
Screenshot 2024-05-30 at 1 32 22 PM

**Step 8**
We dont show the `What we're doing` section when a claim is closed so this what it will look like
Screenshot 2024-05-30 at 12 25 58 PM

## What areas of the site does it impact?

Claim Status Tool

## Acceptance criteria

- [x] There is header text for claim phases 1-7 in the "What We're Doing" component and the number and step description matches exactly what is used as headers on the Overview page.
- [x] The part of the header that mentions the claim "step" matches exactly the name of the step on the Overview page for the new timeline; When a user is on step 1 of 8 the header shows and the description shows
- [x] The description for each phase matches the "What We're Doing" text from [this doc](https://docs.google.com/document/d/1WfWCfoQgKoMBg835VP7j_ptXmqb2o3RVme2u5dMOEr8/edit#heading=h.55axkr7jidx0) consistent with the phase steps.
- [x] There is a link to the overview section of CST that says "Learn more about this step
- [x]  There is "moved to" text that clearly calls out the date the claim moved to this step.
- [x] Behind cst_claim_phases feature flag

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/claims-status/tests/components/OverviewPage.unit.spec.jsx [hunk 1]
```diff
@@ -166,7 +166,7 @@ describe('<OverviewPage>', () => {
     });
   });
 
-  context('cstClaimPhases feature flag enabled', () => {
+  context('cstClaimPhases feature flag disabled', () => {
```
Comment (pmclaren19) [inline src/applications/claims-status/tests/components/OverviewPage.unit.spec.jsx:169]: Noticed this typo and fixed

File: src/applications/claims-status/tests/components/ClaimStatusPage.unit.spec.jsx [hunk 2]
```diff
@@ -74,7 +74,7 @@ describe('<ClaimStatusPage>', () => {
                 claimPhaseDates: {
                   currentPhaseBack: false,
                   phaseChangeDate: '2015-01-01',
-                  latestPhaseType: 'INITIAL_REVIEW',
+                  latestPhaseType: 'UNDER_REVIEW',
```
Comment (pmclaren19) [inline src/applications/claims-status/tests/components/ClaimStatusPage.unit.spec.jsx:77]: INITIAL_REVIEW isnt a real phase type so updated this to a real phase type

File: src/applications/claims-status/utils/helpers.js [hunk 3]
```diff
@@ -43,6 +43,43 @@ export function getStatusDescription(status) {
...
+  GATHERING_OF_EVIDENCE:
+    'We’re reviewing your claim to make sure we have all the evidence and information we need. If we need anything else, we’ll contact you.',
+  REVIEW_OF_EVIDENCE:
+    'We’re reviewing all the evidence for your claim. If we need more evidence or you submit more evidence, your claim will go back to Step 3: Evidence gathering.',
+  PREPARATION_FOR_DECISION:
+    'We’re deciding your claim and determining your disability rating. If we need more evidence or you submit more evidence, your claim will go back to Step 3: Evidence gathering.',
+  PENDING_DECISION_APPROVAL:
+    'We’re preparing your decision letter. If we need more evidence or you submit more evidence, your claim will go back to Step 3: Evidence gathering.',
+  PREPARATION_FOR_NOTIFICATION:
+    'A senior reviewer is doing a final review of your claim and the decision letter. If we need more evidence or you submit more evidence, your claim will go back to Step 3: Evidence gathering.',
```
Comment (samcoforma) [inline src/applications/claims-status/utils/helpers.js:74]: This isn't so much feedback on the pull request as it is a question re: the design. The language here is for step 7, and says that you can go back to Step 3 potentially. However, in the designs I've seen, the diagram shows that you can only go back to Step 3 between steps 4 - 6, not 7. @pmclaren19 @uxgary 

image

**Review by samcoforma — APPROVED** [review]

**Review by ataker — APPROVED** [review]

File: src/applications/claims-status/utils/helpers.js (hunk 3 above)
Comment (pmclaren19) [reply src/applications/claims-status/utils/helpers.js:74]: https://dsva.slack.com/archives/C056854SVT8/p1717168745828499?thread_ts=1717166351.850769&cid=C056854SVT8 @samcoforma Sounds like after somme workshops we learned the claim can revert to step 3 at any step. @uxgary is going to create a new ticket to update the diagram but we can leave this language as is!

**Comment by uxgary** [comment]
Checked the screenshots in this PR. The content in the status page is accurate, however in staging, I was able to review content for steps 2,3,5, and the content in step 2 is incorrect. 

Screenshot 2024-05-31 at 2 05 23 PM

**Comment by pmclaren19** [comment]
> Checked the screenshots in this PR. The content in the status page is accurate, however in staging, I was able to review content for steps 2,3,5, and the content in step 2 is incorrect.
> 
> Screenshot 2024-05-31 at 2 05 23 PM


@uxgary this is what the design in figma had
Screenshot 2024-05-31 at 12 08 46 PM so I was going off of the figma design. Also these changes are part of an entirely different ticket that has already been worked and approved. So if you want step 2 changed we should update the figma design an a bug ticket should be created.

**Pull request merged**